consensus/src/messages/handlers.rs

```diff
@@ -578,10 +579,8 @@ impl RequestTransactionsProof {
                 return Err(ResponseTransactionProofError::BlockNotFound);
             };
 
-            // If it is a checkpoint block, we have some extra work to do
-            if Policy::is_macro_block_at(proving_block_number)
-                && !Policy::is_election_block_at(proving_block_number)
-            {
+            // We have some extra work in the current epoch, if the block we are proving is not our current head
+            if proving_block_number > election_head && proving_block_number < current_head {
```

Comment:
What makes the `current_head` not applicable here?

Reply:
In that case, `verifier_state` can be `None` as it would be without entering this `if` condition